### ISSUE TITLE
Export Custom Properties for EnumProperty Type

### ISSUE THREAD

**Issue by Ziflin** [KhronosGroup/glTF-Blender-IO#730]
It appears that currently "EnumProperty()" types are not exported as Custom Properties? Would this be difficult to add to avoid needing to wrap an EnumProperty with a StringProperty every time?

Attached is an example Blender file with a simple script that adds an "Exported Properties" panel to object properties (just above the Custom Properties panel). This script also adds a "Collider Type" EnumProperty that is not getting exported although the other custom BooleanProperty  for "Collidable" is being exported into "extras."

[EnumProperties.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/3720891/EnumProperties.zip)

**Comment by scurest** [comment]
Although it looks like it does from the UI, the object doesn't have a `colliderShape` custom property just from running the script. You have to click the drop-down and select something for it to be set.

You can test this by typing `D.objects[0]['colliderShape']` into the Python console. It doesn't exist until you select something with the drop-down.

**Comment by julienduroure** [comment]
Yes, confirm that blender maybe come from Blender or from your script, not from exporter, because data can't be retrieve from python until you change it in UI.

**Comment by Ziflin** [comment]
Sorry! I swear I tested changing it thinking that was the problem. There's not a setting to have it export as the enum's string id vs the numeric value I assume? I can just use a getter/setter I think if we really need it to export a string value.

**Comment by scurest** [comment]
So currently the exporter uses the dict-like properties of an object to use for exporting, ie. what you get with `obj['prop']` and what shows up in the "Custom Properties" widget.

You're using custom `bpy.props` properties ie. what you get with `obj.prop`. These appear to store their data in dict-like properties with the same name, which is the only reason they even work at all. When a `bpy.props` property is added, the dict-like property with the same name no longer shows up in the "Custom Properties" widget, and it will re-appear if the `bpy.props` is removed. Also, the dict-like property doesn't get set until you change the custom property (ex. with the drop-down or by using `obj.prop = val`), which is why `colliderShape` wasn't being exported here.

I don't know if we want to do this, but [here's](https://github.com/KhronosGroup/glTF-Blender-IO/compare/master...scurest:export_bpy_props_extras) a patch that will try to handle these kind of `bpy.props` custom properties when exporting. It will pick up the default value of `colliderShape` even if it hasn't been set, and it uses the value of `obj.colliderShape` which is a string, instead of `obj['colliderShape']`, which is a number, so you'd get:

````json
    "extras" : {
        "collidable" : true,
        "colliderShape" : "ConvexHull"
    },
````

**Comment by Ziflin** [comment]
@scurest Yes, I was actually confused as to how the exporter was finding the `bpy.props` ones, but I was glad as they seemed a bit easier to use in a custom panel. Basically we'd like to write a Blender plugin similar to what I posted above and expose properties in a similar manner rather than having to tell people to click "Add" on the "Custom Properties" panel every time. I'm still new to Blender's API so I was mostly going by other examples I'd found for adding properties from Python.

I did a test with the Json reader we're using and the integer value actually works fine, but I mentioned the string (before I removed that comment a minute ago) just because I assumed that's what it would be exporting.

Not writing out the _default_ (unset) value for `bpy.props` properties was actually ok as the Json ends up with less stuff in it similar to how node position/rotation/scale values work with identity values. So maybe your string patch with a check that skips writing the property if the value is unset _or_ if it equals the `bpy.props` default value would be more inline with other glTF properties?
